docs/components/multiselect.md

```diff
@@ -57,6 +60,58 @@ An array of items containing a `label` and `value`. You may also specify:
 />
 ```
 
+### enableItemCreation
+
+`KMultiselect` can offer users the ability to add custom items to the list by typing the item they want to and then clicking the `... (New value)` item at the bottom of the list, which will also automatically select it.
+
+Newly created items will have a `label` consisting of the user input and a randomly generated id for the `value` to ensure uniqueness. It will also have an attribute `custom` set to `true`. This action triggers an `item:added` event containing the added item data.
+
+Deselecting the item will completely remove it from the list and trigger a `item:removed` event containing the removed item's data.
+
+:::tip NOTE
+You cannot add an item if the `label` matches the `label` of a pre-existing item. In that scenario the `... (New value)` item will not be displayed.
+:::
+
+<ClientOnly>
+  <KLabel>Value:</KLabel> <pre class="json ma-0">{{ JSON.stringify(mySelections) }}</pre>
+  <KLabel>Added Items:</KLabel> <pre class="json ma-0">{{ JSON.stringify(addedItems) }}</pre>
+  <KMultiselect
+    v-model="mySelections"
+    :items="deepClone(defaultItems)"
+    enable-item-creation
+    class="mt-2"
+    @item:added="(item) => trackNewItems(item, true)"
+    @item:removed="(item) => trackNewItems(item, false)"
+  />
+</ClientOnly>
+
+```html
+<template>
+  <KLabel>Value:</KLabel> {{ mySelections }}
+  <KLabel>Added Items:</KLabel> {{ addedItems }}
+  <KMultiselect
+    v-model="mySelections"
+    :items="items"
+    enable-item-creation
+    @item:added="(item) => trackNewItems(item, true)"
+    @item:removed="(item) => trackNewItems(item, false)"
+  />
+</template>
+
+<script setup lang="ts">
+  const mySelections = ref([])
```

Comment:
Since the select already has items selected, I think we need to add them here in the example (assuming these are the correct values)
```suggestion
  const mySelections = ref(['cats','bunnies'])
```